File: demo2/src/main/java/org/mosquito/demo2/mapper/OrderMapper.java
package org.mosquito.demo2.mapper;

import org.apache.ibatis.annotations.Mapper;

import org.mosquito.demo2.model.OrderExt;
import org.mosquito.framework.jdbc.mapper.BaseMapper;
import org.mosquito.demo2.model.Order;

import java.util.List;

/**
 * 数据访问接口
 *
 * @author jcchen
 * @Date 2018-5-27
 * 
 */
@Mapper
public interface OrderMapper extends BaseMapper<Order> {

    List<OrderExt> findListExt(Order order);

}

File: demo3/src/main/scala/org/mosquito/demo3/mapper/OrderMapper.scala
package org.mosquito.demo3.mapper

import java.util

import org.apache.ibatis.annotations.Mapper
import org.mosquito.demo3.model.{Order, OrderExt}
import org.mosquito.framework.jdbc.mapper.BaseMapper

@Mapper
trait OrderMapper extends BaseMapper[Order] {

  def findListExt(order: Order): util.List[OrderExt]

  def findListScala(order: Order): util.List[Order]
}
